src/main/java/org/wso2/carbon/inbound/smpp/SMPPListeningConsumer.java

```diff
@@ -136,6 +148,7 @@
      * An object that executes submitted Runnable tasks.
      */
     private RetryExecutor executor;
+    private MessageContext msgCtx;
```

Comment:
Can you pass the MessageContext to injectMessage method as a parameter. This is not a good way of doing this via a class level variable.